pallets/hyperdrive/src/tezos.rs

```diff
@@ -0,0 +1,570 @@
+/// ```
+fn parse_job_registration_payload<Reward, Balance, ParsableAccountId, AccountId, Extra>(
+    encoded: &[u8],
+) -> Result<(JobIdSequence, JobRegistration<AccountId, Extra>), ValidationError>
+where
+    ParsableAccountId: FromStr + Into<AccountId>,
+    Extra: From<RegistrationExtra<Reward, Balance, AccountId>>,
+    Reward: Parameter + Member + TryFrom<Vec<u8>>,
+    Balance: From<u128>,
+{
+    let unpacked: Micheline = Micheline::unpack(encoded, Some(registration_payload_schema()))
+        .map_err(|e| ValidationError::TezosMicheline(e))?;
+
+    let p: PrimitiveApplication = unpacked.try_into()?;
+    let pair: Pair = p.try_into()?;
+
+    let values = pair.flatten().values;
+    if values.len() != 18 {
+        Err(ValidationError::InvalidMessage)?;
+    }
```

Comment:
This is already checked by the unpacking.
```suggestion
```

Reply:
it probably does now, but will it do in all future?

Reply:
Yes, the bytes need to match the type.